loader.c

```diff
@@ -122,6 +127,11 @@ loader_main(struct loader_callbacks *cb, void *arg, int version, int ndisks)
         if (multiboot_load(kernel, kernsz, mb)) {
             goto error;
         }
+
+        if (multiboot_info_set_cmdline(&mb->info, loader_args.cmdline)) {
```

Comment:
This should really only be called if `loader_args.cmdline` is set, otherwise we segfault here, as it is initialized to `NULL`.